File: native/gameplugin/scripting/lua-bindings/auto/api/IETAnalyticHelper.lua

--------------------------------
-- @module IETAnalyticHelper
-- @parent_module ccg

--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] purchase 
-- @param self
-- @param #string name
-- @param #int amount
-- @param #double coin
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] use 
-- @param self
-- @param #string name
-- @param #int amount
-- @param #double coin
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] missionStart 
-- @param self
-- @param #string missionId
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] missionFailed 
-- @param self
-- @param #string missionId
-- @param #string reason
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] setLevel 
-- @param self
-- @param #int level
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] charge 
-- @param self
-- @param #string name
-- @param #double cash
-- @param #double coin
-- @param #int type
        
--------------------------------
-- @overload self, string, string         
-- @overload self, string         
-- @overload self, string, map_table         
-- @function [parent=#IETAnalyticHelper] onEvent
-- @param self
-- @param #string eventId
-- @param #map_table map

--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] setAccountInfo 
-- @param self
-- @param #map_table userInfo
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] reward 
-- @param self
-- @param #double coin
-- @param #int type
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] missionSuccess 
-- @param self
-- @param #string missionId
        
--------------------------------
-- 
-- @function [parent=#IETAnalyticHelper] getInstance 
-- @param self
-- @return IETAnalyticHelper#IETAnalyticHelper ret (return value: IETAnalyticHelper)
        
return nil

File: native/gameplugin/android/IETAnalyticHelper.cpp
//
//  IETAnalyticHelper.mm
//  gameplugin
//
//  Created by geekgy on 15/6/11.
//
//

#include "IETAnalyticHelper.h"
#include "IETAndroidBridge.h"

using namespace std;
using namespace cocos2d;

#define ANALYTIC_HELPER_CLASS_NAME "com.joycastle.gameplugin.AnalyticHelper"

void IETAnalyticHelper::setAccountInfo(cocos2d::ValueMap userInfo)
{
    ValueVector reqVec;
    reqVec.push_back(Value(userInfo));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME, "setAccoutInfo", reqVec);
}

void IETAnalyticHelper::onEvent(std::string eventId)
{
    ValueVector reqVec;
    reqVec.push_back(Value(eventId));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"onEvent",reqVec);
}

void IETAnalyticHelper::onEvent(std::string eventId, std::string label)
{
    ValueVector reqVec;
    reqVec.push_back(Value(eventId));
    reqVec.push_back(Value(label));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"onEvent",reqVec);
}

void IETAnalyticHelper::onEvent(std::string eventId, cocos2d::ValueMap map)
{
    ValueVector reqVec;
    reqVec.push_back(Value(eventId));
    reqVec.push_back(Value(map));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME, "onEvent", reqVec);
}

void IETAnalyticHelper::setLevel(int level)
{
    ValueVector reqVec;
    reqVec.push_back(Value(level));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME, "setLevel", reqVec);
}

void IETAnalyticHelper::charge(std::string name, double cash, double coin, int type)
{
    ValueVector reqVec;
    reqVec.push_back(Value(name));
    reqVec.push_back(Value(cash));
    reqVec.push_back(Value(coin));
    reqVec.push_back(Value(type));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME, "charge", reqVec);
}

void IETAnalyticHelper::reward(double coin, int type)
{
    ValueVector reqVec;
    reqVec.push_back(Value(coin));
    reqVec.push_back(Value(type));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"reward",reqVec);
}

void IETAnalyticHelper::purchase(std::string name, int amount, double coin)
{
    ValueVector reqVec;
    reqVec.push_back(Value(name));
    reqVec.push_back(Value(amount));
    reqVec.push_back(Value(coin));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"purchase",reqVec);
}

void IETAnalyticHelper::use(std::string name, int amount, double coin)
{
    ValueVector reqVec;
    reqVec.push_back(Value(name));
    reqVec.push_back(Value(amount));
    reqVec.push_back(Value(coin));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"use",reqVec);
}

void IETAnalyticHelper::missionStart(std::string missionId)
{
    ValueVector reqVec;
    reqVec.push_back(Value(missionId));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"onMissionBegin",reqVec);
}

void IETAnalyticHelper::missionSuccess(std::string missionId)
{
    ValueVector reqVec;
    reqVec.push_back(Value(missionId));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"onMissionCompleted",reqVec);
}

void IETAnalyticHelper::missionFailed(std::string missionId, std::string reason)
{
    ValueVector reqVec;
    reqVec.push_back(Value(missionId));
    reqVec.push_back(Value(reason));
    IETAndroidBridge::getInstance()->callJavaMethod(ANALYTIC_HELPER_CLASS_NAME,"onMissionFailed",reqVec);
}
